### DISCUSSION TITLE
how can I get a scanner data (stocks) from a preset one?

### DISCUSSION THREAD

**Discussion by s-badran** [stoqey/ib#197] [General]
I need to select the stock dynamically from a previously set scanner on TWS. I can't find information on how to do this using this package. Is it a not implemented feature, or am I missing where to find it?? 
Either way, any help would be appreciated.

Thank you

**Comment by rylorin** [comment]
Hello,
May be I misunderstood but if you subscribe to `getMarketScanner` you will get results as `MarketScannerUpdate` which contains all contract (stock) information. Isn't it what you need?
Regards

**Reply by s-badran** [reply]
Thanks for the response. Can you please refer me to where I can find examples on how to do this? I looked at the documentation and could find one function **_ib.reqScannerSubscription_** but still don't get exactly how to use it (e.g. what is the request id or tickerId? and where to get it from? same thing about other parameters)

**Reply by s-badran** [reply]
Here is a further elaboration: the best thing I could come up with is the following:
```
/* eslint-disable @typescript-eslint/no-explicit-any */
// ScannerOperations.ts

import {IBApi } from '@stoqey/ib';
import { Instrument, ScannerSubscription, LocationCode, ScanCode} from '@stoqey/ib';

export class ScannerOperations {
  private ib: IBApi;

  constructor(ib: IBApi) {
    this.ib = ib;
  }

  async fetchTopStocks(
    scannerName: string,
    topStocksCount: number
  ): Promise<string[]> {
    const scannerParams: ScannerSubscription = {
      instrument: Instrument.STK,
      locationCode: LocationCode.STK_US_MAJOR,
      scanCode: ScanCode.ALL_SYMBOLS_ASC,
      numberOfRows: topStocksCount,
    };
    const scannerData = this.ib.reqScannerSubscription(0, scannerParams);
    return this.extractTopStocks(scannerData, topStocksCount);
  }

  private extractTopStocks(scannerData: any, topStocksCount: number): string[] {
    const sortedStocks = scannerData.sort(
      (a: any, b: any) => b.someMetric - a.someMetric
    );
    return sortedStocks.slice(0, topStocksCount).map((stock: any) => stock.symbol);
  }
}
```
But the **ib.reqScannerSubscription** function only reads from TWS predefined scanners. I have created a **CUSOM** scanner called _LowFloatStocks_ but I don't know where to use it in above code (if everything else is correct)
